solr/modules/ltr/src/java/org/apache/solr/ltr/response/transform/LTRFeatureLoggerTransformerFactory.java

```diff
@@ -241,11 +243,23 @@ public void setContext(ResultContext context) {
      *
      * @param transformerFeatureStore the explicit transformer feature store
      */
-    private LoggingModel createLoggingModel(String transformerFeatureStore) {
+    private LoggingModel createLoggingModel(
+        String transformerFeatureStore, Boolean docsWereNotReranked) {
       final ManagedFeatureStore fr = ManagedFeatureStore.getManagedFeatureStore(req.getCore());
 
+      if (transformerFeatureStore != null) {
+        fr.checkMissingFeatureStore(transformerFeatureStore);
+      }
+
       final FeatureStore store = fr.getFeatureStore(transformerFeatureStore);
 
+      if (docsWereNotReranked) {
```

Comment:
is this because in reranking we add the store to subsequently add the features?

Reply:
This is because I need to differentiate between the two cases where I do not pass the feature store name (i.e., it is null). If you don't pass a reranking query (qr), it means you don't pass the model name either, so the only feature store I could use is the default feature store (which should not be empty, otherwise you get the error).
On the other hand, if I use the reranking query, it means that I have specified the model name, so I temporarily create an empty feature store to later add features in the implTransform() method, using the methods of the FeatureLogger class.